crates/ar-cli/src/db.rs

```diff
@@ -0,0 +1,151 @@
+use core::fmt;
+use std::path::PathBuf;
+
+use rusqlite::{Connection, Result, Row};
+
+pub fn init(path: &PathBuf) -> Result<Connection, rusqlite::Error> {
+    Connection::open(path)
+}
+
+#[derive(Debug)]
+pub struct Event {
+    pub tx_hash: Vec<u8>,
+    pub block_number: u64,
+    pub mac_salt: Vec<u8>,
+    pub mac_commitment: Vec<u8>,
+    pub viewing_key: Option<Vec<u8>>,
+}
+
+impl fmt::Display for Event {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let tx_hash_hex = hex::encode(&self.tx_hash);
+        let mac_salt_hex = hex::encode(&self.mac_salt);
+        let mac_commitment_hex = hex::encode(&self.mac_commitment);
+        let viewing_key_hex = self
+            .viewing_key
+            .as_ref()
+            .map_or("None".to_string(), hex::encode);
+        write!(
+            f,
+            "Event {{\n  tx_hash: 0x{},\n  block_number: {},\n  mac_salt: 0x{},\n  mac_commitment: 0x{},\n  viewing_key: 0x{}\n}}",
+            tx_hash_hex, self.block_number, mac_salt_hex, mac_commitment_hex, viewing_key_hex
+        )
+    }
+}
+
+pub fn create_events_table(connection: &Connection) -> Result<(), rusqlite::Error> {
+    connection.execute(
+        "CREATE TABLE IF NOT EXISTS events (
+            tx_hash BLOB PRIMARY KEY,
+            block_number INTEGER NOT NULL,
+            mac_salt BLOB NOT NULL,
+            mac_commitment BLOB NOT NULL,
+            viewing_key BLOB
```

Comment:
1. do we need/want to connect it to the other table?
2. this can be null, because initially, when we fetch an event, we don't know the viewing key, right?

Reply:
after full review, I can answer these questions myself, resolving